frontend/src/reducer/user.js

```diff
@@ -33,25 +48,27 @@ export default function user (state = defaultUser, action) {
     case `${SET}/${USER_CONNECTED}`:
       return {
         ...state,
-        ...action.user,
-        lang: action.user.lang ? action.user.lang : state.lang
+        ...serializeUser(action.user),
+        lang: serializeUser(action.user).lang ? serializeUser(action.user).lang : state.lang
       }
 
     case `${SET}/${USER_DISCONNECTED}`:
-      return { ...defaultUser, lang: state.lang, logged: false }
+      return { ...state, lang: state.lang, logged: false }
 
     case `${SET}/${USER_LANG}`:
       return { ...state, lang: action.lang }
 
-    case `${UPDATE}/${USER_PUBLIC_NAME}`:
-      return { ...state, public_name: action.newName }
+    case `${UPDATE}/${USER}`:
+      return {
+        ...state,
+        ...serializeUser(action.newUser),
+        agendaUrl: serializeUser(action.newUser).agendaUrl ? serializeUser(action.newUser).agendaUrl : state.agendaUrl,
+        logged: serializeUser(action.newUser).logged ? serializeUser(action.newUser).logged : state.logged
+      }
```

Comment:
avoid multiple calls to the same function and try using `||` for ternaries that returns the condition
```suggestion
    case `${UPDATE}/${USER}`:
      const serializedUser = serializeUser(action.newUser)
      return {
        ...state,
        ...serializedUser,
        agendaUrl: serializedUser.agendaUrl || state.agendaUrl,
        logged: serializedUser.logged || state.logged
      }
```

Reply:
Changed